images/base/Dockerfile

```diff
@@ -117,44 +120,48 @@ RUN git clone --filter=tree:0 "${CONTAINERD_CLONE_URL}" /containerd \
     && cd /containerd \
     && git checkout "${CONTAINERD_VERSION}" \
     && eval "$(gimme "${GO_VERSION}")" \
+    && export GOARCH=$TARGETARCH && export CC=$(target-cc) && export CGO_ENABLED=1 \
     && make bin/ctr bin/containerd bin/containerd-shim-runc-v2 \
     && go-licenses save --save_path=/_LICENSES \
         ./cmd/ctr ./cmd/containerd ./cmd/containerd-shim-runc-v2
 
 # stage for building runc
 FROM go-build as build-runc
-ARG GO_VERSION
+ARG TARGETARCH GO_VERSION
 ARG RUNC_VERSION="v1.1.7"
 ARG RUNC_CLONE_URL="https://github.com/opencontainers/runc"
 RUN git clone --filter=tree:0 "${RUNC_CLONE_URL}" /runc \
     && cd /runc \
     && git checkout "${RUNC_VERSION}" \
     && eval "$(gimme "${GO_VERSION}")" \
+    && export GOARCH=$TARGETARCH && export CC=$(target-cc) && export CGO_ENABLED=1 \
     && make runc \
     && go-licenses save --save_path=/_LICENSES .
```

Comment:
not sure if go-licenses is GOARCH aware, we might want to set GOARCH=$TARGETARCH for each of these

Reply:
done. it probably is indirectly via whatever it uses to list dependency packages.